graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -263,6 +269,24 @@ public GraphqlServiceBuilder useBlockingTaskExecutor(boolean useBlockingTaskExec
         return this;
     }
 
+    /**
+     * Sets whether [[GraphQL over WebSocket Protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md)] is supported.
+     */
+    public GraphqlServiceBuilder useWebSocket(boolean useWebSocket) {
+        this.useWebSocket = useWebSocket;
+        return this;
+    }
+
+    /**
+     * Sets an optional {@link WebSocketServiceBuilder} customizer.
+     */
+    public GraphqlServiceBuilder webSocketBuilderCustomizer(
+            Consumer<WebSocketServiceBuilder> webSocketBuilderCustomizer) {
+        checkState(useWebSocket, "useWebSocket must be true to customize WebSocketServiceBuilder");
```

Comment:
Do you mind checking this at `build()` instead? This check may pass if the user sets `useWebSocket == true` afterwards.

```
public GraphqlService build() {
...
checkArgument(useWebSocket || webSocketBuilderCustomizer == null,
                      "useWebSocket must be true to customize WebSocketServiceBuilder");
```